frame/staking-rewards/src/lib.rs

```diff
@@ -494,11 +515,80 @@ pub mod pallet {
 
 		#[transactional]
 		fn unstake(
-			_who: &Self::AccountId,
-			_position: &Self::PositionId,
+			who: &Self::AccountId,
+			position_id: &Self::PositionId,
 			_remove_amount: Self::Balance,
 		) -> DispatchResult {
-			Err("Not implemented".into())
+			let keep_alive = false;
+			let stake = Stakes::<T>::try_get(position_id).map_err(|_| Error::<T>::StakeNotFound)?;
+			ensure!(who == &stake.owner, Error::<T>::OnlyStakeOwnerCanUnstake);
+			ensure!(!stake.reductions.is_empty(), Error::<T>::ReductionConfigProblem);
+			let early_unlock = stake.lock.started_at.safe_add(&stake.lock.duration)? >=
+				T::UnixTime::now().as_secs();
+			let pool_id = stake.reward_pool_id;
+			let mut rewards_pool =
+				RewardPools::<T>::try_get(pool_id).map_err(|_| Error::<T>::RewardsPoolNotFound)?;
+
+			let mut inner_rewards = rewards_pool.rewards.into_inner();
```

Comment:
afaik you do not need use inner. perbills and fixed are composable in substrate

Reply:
`rewards_pool.rewards` is [BoundedBTreeMap](https://paritytech.github.io/substrate/master/sp_runtime/bounded/bounded_btree_map/struct.BoundedBTreeMap.html).